File: tests/parsers/fixtures/sample.cpp
// Sample C++ source for parser testing.

#include <string>
#include <vector>
#include <algorithm>

namespace sample {

/// Processes data records.
class DataProcessor {
public:
    DataProcessor(const std::string& config) : config_(config) {}

    std::vector<std::string> process(const std::vector<std::string>& data) {
        std::vector<std::string> result;
        result.reserve(data.size());
        for (const auto& item : data) {
            result.push_back(transform(item));
        }
        return result;
    }

private:
    std::string config_;

    std::string transform(const std::string& item) {
        std::string out = item;
        out.erase(0, out.find_first_not_of(' '));
        out.erase(out.find_last_not_of(' ') + 1);
        std::transform(out.begin(), out.end(), out.begin(), ::tolower);
        return out;
    }
};

}  // namespace sample

File: tests/parsers/fixtures/sample.swift
/// Sample Swift module for parser testing.

import Foundation

/// Protocol for processable items.
protocol Processable {
    func process(data: [String]) -> [String]
}

/// Processes data records.
class DataProcessor: Processable {
    let config: String

    init(config: String) {
        self.config = config
    }

    func process(data: [String]) -> [String] {
        return data.map { transform(item: $0) }
    }

    private func transform(item: String) -> String {
        return item.trimmingCharacters(in: .whitespaces).lowercased()
    }
}

func loadData(from path: String) -> [String] {
    let content = try? String(contentsOfFile: path)
    return content?.components(separatedBy: "\n") ?? []
}
